### PR TITLE
add 'ability_id_active' attribute to filter

### PR DESCRIPTION
**Pull request by newfrenchy83**
until now, only the tpe of ability could be filtered with activity, and id only for unit who have ability, same if she's inactive.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Seems fine to me.

**Comment by Pentarctagon** [comment]
A WML unit test for this would be good.

**Comment by newfrenchy83** [comment]
> 
> 
> A WML unit test for this would be good.

Done, i await checking to be complete

**Comment by CelticMinstrel** [comment]
Those unit tests don't look right, and they're definitely misnamed. You're not testing that filters on an event work correctly, you're testing that filters themselves work; so the general logic should be:

1. Create a unit with the ability
2. Assert [have_unit] with the particular filter

And the tests should not have "event" in the name.

**Comment by newfrenchy83** [comment]
> 
> 
> Those unit tests don't look right, and they're definitely misnamed. You're not testing that filters on an event work correctly, you're testing that filters themselves work; so the general logic should be:
> 
>     1. Create a unit with the ability
> 
>     2. Assert [have_unit] with the particular filter
> 
> 
> And the tests should not have "event" in the name.

I fixed this.

File: data/test/scenarios/events-test_filters.cfg [hunk 1]
```diff
@@ -148,6 +148,157 @@
...
+        name=start
+        {VARIABLE triggers 0}
+        [unit]
+            id=alex
+            name=_"Alex"
+            x,y=12,4
+            type=Elvish Archer
+            side=1
+        [/unit]
+        [object]
```
Comment (CelticMinstrel) [inline data/test/scenarios/events-test_filters.cfg:162]: This needs to have a `[filter]` tag to match the unit it applies to.

You can probably just use `id=alice` or `id=bob`, no need to create a third unit, is there?

File: data/test/scenarios/events-test_filters.cfg [hunk 2]
```diff
@@ -148,6 +148,157 @@
...
+            silent=yes
```
Comment (CelticMinstrel) [inline data/test/scenarios/events-test_filters.cfg:234]: This also needs a `[filter]`  tag.

**Review by CelticMinstrel — COMMENTED** [review]
That looks better as a unit test, but now you need to move it to a more appropriate file. This unit test does not belong in events-test_filters.cfg.

**Comment by Pentarctagon** [comment]
Also, for the tests that start with `swarm_disables_upgrades_with_abilities`, please update the description so that they say what they're doing rather than being copies of each other.  Either a separate commit here or a separate PR is fine.

**Comment by newfrenchy83** [comment]
> 
> 
> Also, for the tests that start with `swarm_disables_upgrades_with_abilities`, please update the description so that they say what they're doing rather than being copies of each other. Either a separate commit here or a separate PR is fine.

description updated in this PR

**Comment by CelticMinstrel** [comment]
That's better.

**Comment by newfrenchy83** [comment]
@Pentarctagon don't forget this PR please.

File: src/units/filter.cpp [hunk 3]
```diff
@@ -386,6 +408,42 @@ void unit_filter_compound::fill(vconfig cfg)
...
+							return true;
+						}
+					}
+
+					const auto adjacent = get_adjacent_tiles(args.loc);
+					for(unsigned i = 0; i < adjacent.size(); ++i) {
+						const unit_map::const_iterator it = units.find(adjacent[i]);
+						if (it == units.end() || it->incapacitated())
+							continue;
+						if ( &*it == (args.u.shared_from_this()).get() )
```
Comment (Pentarctagon) [inline src/units/filter.cpp:431]: Please remove the extra spaces, so it's `if (&*it == (args.u.shared_from_this()).get())` to keep this consistent.

Comment (newfrenchy83) [reply src/units/filter.cpp]: It done

**Pull request merged**